application-ckeditor-ui/src/main/resources/CKEditor/FileUploader.xml

```diff
@@ -89,11 +90,30 @@
       #end
       #createAttachment($fileName)
     #else
-      #sendError(403 'The CSRF token is missing.')
+      #sendError(403 "$services.localization.render('ckeditor.upload.error.csrf')")
     #end
   #else
     ## Bad Request
-    #sendError(400 "The 'upload' request parameter is missing. Either you didn't submit any file or the size of the submitted file exceeds the configured maximum upload size.")
+    #sendError(400 "$services.localization.render('ckeditor.upload.error.emptyreturn')")
+  #end
+#end
+
+## New way of handling attachment without saving immediately the document.
+#macro (handleTemporaryUploadRequest)
+  #if ($services.csrf.isTokenValid($request.form_token))
+    #set ($document = $doc)
+    #if ("$!request.document" != '')
+      #set ($document = $xwiki.getDocument($request.document))
+    #end
+    #set ($reference = $document.documentReference)
+    #set ($attachment = $services.temporaryattachments.uploadTemporaryAttachment($reference, 'upload'))
```

Comment:
Why lowercase ``temporaryattachments``? Why not camelcase?

```suggestion
    #set ($attachment = $services.temporaryAttachments.uploadTemporaryAttachment($reference, 'upload'))
```

Reply:
Hmmm, sounds like we have a mix of camelcase and lowercase script services, and I'm more used to use the lowercase ones, so I always use lowercase, e.g:
  * $services.officeimporter
  * $services.modelvalidation
  * $services.linkchecker
 
The only two script service using camelcase I can see are:
  * $services.liveData
  * $services.messageStream

Not sure we have a rule for this.

Reply:
Ok tbh I don't have a strong opinion on this, so I will change if you really prefer. In the meantime, I'm opening a discussion thread on the forum to have a more clear rule for this in the future.

Reply:
for info: https://forum.xwiki.org/t/rules-for-script-service-naming/10138